app-backend/database/src/main/scala/com/azavea/rf/database/tables/Scenes.scala

```diff
@@ -401,11 +403,28 @@ object Scenes extends TableQuery(tag => new Scenes(tag)) with LazyLogging {
     * @param user    Results will be limited to user's organization
     */
   def deleteScene(sceneId: UUID, user: User)(implicit database: DB): Future[Int] = {
-    database.db.run {
+    val sceneQuery =
       Scenes
         .filterToSharedOrganizationIfNotInRoot(user)
         .filter(_.id === sceneId)
-        .delete
+    database.db.run {
+      sceneQuery.result
+    } flatMap { (scenes) =>
+      scenes.headOption match {
+        case Some(scene) =>
+          database.db.run {
+            sceneQuery
+              .filterToOwnerIfNotInRootOrganization(user)
+              .delete
+          } map { numDeleted =>
+            numDeleted match {
+              case 0 =>
+                throw IllegalRequestException(StatusCodes.ClientError(403)("Forbidden", "Error deleting scene: not authorized to modify this scene"))
```

Comment:
It makes me a little uncomfortable to have the `delete` sitting between this and the `filterToOwnerIfNotInRootOrganization`. I can't immediately think of a case where the scene would be owned by the user but wouldn't be able to be deleted, but if that ever occurs then this will be confusing. I'm not sure if there's an easy way, but if it's possible to check the user's ownership of the scene _before_ deleting, that'd be my preference.